### DISCUSSION TITLE
Save and use Oauth token

### DISCUSSION THREAD

**Discussion by subhrajyoti21** [Byron/google-apis-rs#303] [Q&A] [answered]
Hey there, how do i get the token that has been generated after Oauth. I want to re use the token for the next set of requests to fetch emails sent to a specific email address. Also, how do i use the access token that has been generated. Below i have updated as per the documentation, but it does not seem to work.

Here's my code. 

```
let secret = yup_oauth2::read_application_secret(std::path::Path::new("./secret.json"))
    .await
    .expect("./secret.json");

let auth = yup_oauth2::InstalledFlowAuthenticator::builder(
        secret,
        yup_oauth2::InstalledFlowReturnMethod::HTTPRedirect,
    )
    .build()
    .await
    .unwrap()

let mut hub = Gmail::new(
        hyper::Client::builder().build(hyper_rustls::HttpsConnector::with_native_roots()),
        auth,
    );

let result = hub
        .users()
        .messages_list("userId")
        .user_id("id”)
        .param("oauth_token", "<token>“)
        .param("access_token", "<token>“)
        .max_results(55)
        .doit()
        .await;
```

Please help.

**Comment by Byron** [comment, marked as answer]
This actually wasn't easily possible after the `auth` instance was handed to the `hub`.
I have fixed that to allow access to `hub.auth`, allowing to call `hub.auth.token()`, which is also done internally. It's available in `gmail1-v2.0.9`.

**Reply by subhrajyoti21** [reply]
Thanks @Byron for the update

**Reply by subhrajyoti21** [reply]
A follow up question on this - 
How do i use the acces token retrieved from `hub.auth.token()`?
Basically, once the token is obtained a i should not require user approval to fetch their emails

**Reply by Byron** [reply]
It's important to set the correct scopes when obtaining authorization for these. Furthermore this library adds the token to each request depending on the required scopes for that request. There should be no need to add own parameters for authorization.
